chaindataloader/sideload.go

```diff
@@ -0,0 +1,73 @@
+package chaindataloader
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+
+	// ErrEndOfFile is returned when we have reached the end of a file while reading it.
+	ErrEndOfFile = sideLoadErr("end of file reached")
+)
+
+// sideLoadErr is a type that represents the defined errors in the chaindataloader package.
+type sideLoadErr string
+
+// Error returns the error in string form.
+func (e sideLoadErr) Error() string {
+	return string(e)
+}
+
+// Reader is the interface a caller fetches data from.
+type Reader[messageType any | *wire.BlockHeader] interface {
```

Comment:
See about about making this a batch interface. 

Also I don't think the type parameters are really being used for anything here? So then can be dropped, since we only ever care about using this for headers. Unless you plan to also use it for filter headers? 

Either way, for a type param constraint, the type constraint should always be an interface.

Reply:
So my first attempt was to make that an interface as `wire.BlockHeader` and `*wire.MsgCFHeaders` don't implement an interface that could not work, I will include the `*wire.MsgCFHeaders` in the reader in the next PR that sideloads the filter headers

Reply:
Ah then in that case, I think we can make a new interface that abstracts over both of them. Or you can just have the type constraint be: `wire.BlockHeader | wire.MsgCFHeaders`.